pkg/ccl/backupccl/restore_job.go

```diff
@@ -1696,6 +1770,15 @@ func (r *restoreResumer) publishDescriptors(
 		if err := checkVersion(mutTable, tbl.Version); err != nil {
 			return err
 		}
+		badIndexes := devalidateIndexes[mutTable.ID]
+		for _, badIdx := range badIndexes {
+			found, err := mutTable.FindIndexWithID(badIdx)
+			if err != nil {
+				return err
+			}
+			mutTable.RemovePublicNonPrimaryIndex(found.Ordinal())
+			mutTable.AddIndexMutation(found.IndexDesc(), descpb.DescriptorMutation_ADD)
+		}
```

Comment:
@ajwerner I occasionally see ` duplicate: index "balance_idx_4" in the middle of being added, not yet public` when I get to the write below. Any obvious mistake here?

Reply:
huh, something is weird. I added printlns of AllNonDropIndexes before and after the remove/add pair and see this: 

```
tbl 54 idx 1 "primary" (adding: false)
tbl 54 idx 2 "balance_idx_1" (adding: false)
tbl 54 idx 3 "balance_idx_2" (adding: false)
tbl 54 idx 4 "balance_idx_3" (adding: false)
tbl 54 idx 5 "balance_idx_4" (adding: false)
removing idx 4 (in pos 3) from tbl 54 and re-adding as "balance_idx_3"
 -> tbl 54 idx 1 "primary" (adding: false)
 -> tbl 54 idx 2 "balance_idx_1" (adding: false)
 -> tbl 54 idx 3 "balance_idx_2" (adding: false)
 -> tbl 54 idx 5 "balance_idx_4" (adding: false)
 -> tbl 54 idx 5 "balance_idx_4" (adding: true)
```
that is unexpected -- we moved 3 (idx_4) and re-added it, but we see 5 duplicated instead